src/components/MessageList/utils/getGroupStyles.ts

```diff
@@ -39,42 +45,46 @@ export const getGroupStyles = <
 >(
   params: GetGroupStylesParams<At, Ch, Co, Ev, Me, Re, Us>,
 ) => {
-  const { messages, noGroupByUser } = params;
+  const {
+    dateSeparators,
+    hideDateSeparators,
+    maxTimeBetweenGroupedMessages,
+    messages,
+    noGroupByUser,
+    userId,
+  } = params;
   const messageGroupStyles: { [key: string]: GroupType[] } = {};
 
-  for (let i = 0; i < messages.length; i++) {
-    const previousMessage = messages[i - 1];
-    const message = messages[i];
-    const nextMessage = messages[i + 1];
-    const groupStyles: GroupType[] = [];
+  const messagesFilteredForNonUser = messages.filter(
+    (message) => !message.deleted_at || userId === message.user?.id,
+  );
 
-    /**
-     * Skip event and date messages
-     */
-    if (message.type === 'channel.event') {
-      continue;
-    }
+  for (let i = 0; i < messagesFilteredForNonUser.length; i++) {
+    const previousMessage = messagesFilteredForNonUser[i - 1];
```

Comment:
on the first iteration of this loop won't this result in a `-1` index check and crash?

Reply:
It will be -1 but that will just be undefined, same with the last loop with +1; could make the types reflect that though lemme look at getting that right.

Reply:
Added a casting to correct for possibly undefined, this is a known TS thing, we've discussed it before too ;) :https://github.com/microsoft/TypeScript/issues/13778